tasks/push-rpm-manifests-to-pyxis/push-rpm-manifests-to-pyxis.yaml

```diff
@@ -0,0 +1,155 @@
+      description: The workspace where the snapshot spec json file resides
+  volumes:
+    - name: workdir
+      emptyDir: {}
+  steps:
+    - name: download-sbom-files
+      image:
+        quay.io/redhat-appstudio/release-service-utils:8bf56a04aaeb371f4a822d2b76520e9bdcacfb26
+      volumeMounts:
+        - mountPath: /workdir
+          name: workdir
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        PYXIS_FILE="$(workspaces.data.path)/$(params.pyxisJsonPath)"
+        if [ ! -f "${PYXIS_FILE}" ] ; then
+            echo "No valid pyxis file was provided."
+            exit 1
+        fi
+
+        IMAGEURLS=($(cat "${PYXIS_FILE}" | jq -r '.components[].pyxisImages[].containerImage' | sort | uniq))
+
+        mkdir /workdir/sboms
+        cd /workdir/sboms
+
+        for i in ${!IMAGEURLS[@]}; do
+          echo "Fetching sbom for image: ${IMAGEURLS[$i]}"
+          cosign download sbom --output-file "${IMAGEURLS[$i]}.json" "${IMAGEURLS[$i]}"
+        done
+
+        SBOM_COUNT=$(ls *.json | wc -l )
+        if [ $SBOM_COUNT != ${#IMAGEURLS[@]} ]; then
+          echo "ERROR: Expected to fetch sbom for ${#IMAGEURLS[@]} images, but only $SBOM_COUNT were saved"
+          exit 1
+        fi
+
+    - name: push-rpm-manifests-to-pyxis
+      image:
+        quay.io/redhat-appstudio/release-service-utils:8bf56a04aaeb371f4a822d2b76520e9bdcacfb26
+      env:
+        - name: pyxisCert
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: cert
+        - name: pyxisKey
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: key
+      volumeMounts:
+        - mountPath: /workdir
+          name: workdir
+      script: |
+        #!/usr/bin/env sh
+        set -eu
+
+        if [[ "$(params.server)" == "production" ]]
+        then
+          export PYXIS_GRAPHQL_API="https://graphql-pyxis.api.redhat.com/graphql/"
+        elif [[ "$(params.server)" == "stage" ]]
+        then
+          export PYXIS_GRAPHQL_API="https://graphql-pyxis.preprod.api.redhat.com/graphql/"
+        else
+          echo "Invalid server parameter. Only 'production' and 'stage' are allowed."
+          exit 1
+        fi
+
+        export PYXIS_CERT_PATH=/workdir/crt
+        export PYXIS_KEY_PATH=/workdir/key
+        echo "${pyxisCert}" > $PYXIS_CERT_PATH
+        echo "${pyxisKey}" > $PYXIS_KEY_PATH
+
+        PYXIS_FILE="$(workspaces.data.path)/$(params.pyxisJsonPath)"
+
+        cd /workdir/sboms
+
+        N=$(params.concurrentLimit)  # The maximum number of images to be processed at once
+        declare -a jobs=()
+        declare -a files=()
+        total=$(ls *.json | wc -l )
+        count=0
+        success=true
+        echo "Starting RPM Manifest upload for $total files in total. " \
+          "Up to $N files will be uploaded at once..."
+
+        for IMAGE in $(jq -r '.components[].pyxisImages[]' "${PYXIS_FILE}"); do
```

Comment:
I would recommend iterating over indexes - at the very least you should use `-c` when iterating over json strings, but it's still prone to error if there is a string with a space somewhere inside the json - probably not the case here, but doesn't seem safe. (Also, the `-r` is redundant here.)

Like this:
```sh
# Get the length of the array
array_length=$(jq '. | length' "$JSON_FILE")

# Iterate over each index of the array
for ((index = 0; index < array_length; index++)); do
    # Get the item at the current index
    item=$(jq -r --argjson index "$index" '.[$index]' "$JSON_FILE")
```
You would have two nested for loops.

Reply:
I'm getting `jq: error (at /workspace/data/pyxis.json:51): Cannot index object with number`. Any ideas?

Reply:
I think it's ok :)